### PR TITLE
flite: bump to 2.1

### PR DESCRIPTION
**Pull request by alwayslivid**
Additionally, I also recreated the patchset that was originally made by @TimothyGu.

### CODE REVIEW

File: app-accessibility/flite/patches/flite-2.1.patchset [hunk 1]
```diff
@@ -0,0 +1,67 @@
+From fb6af52e85a36505056b10ca04647e5aacdfd36a Mon Sep 17 00:00:00 2001
+From: Panagiotis Vasilopoulos <hello@alwayslivid.com>
```
Comment (Begasus) [inline app-accessibility/flite/patches/flite-2.1.patchset]: Keep credentials from original author

File: app-accessibility/flite/flite-2.1.recipe [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+# Although individually many files have different licenses, as a whole this
+# license covers all the files.
+LICENSE="Flite"
+REVISION="3"
+SOURCE_URI="http://www.festvox.org/flite/packed/flite-2.1/flite-${portVersion}-release.tar.bz2"
+CHECKSUM_SHA256="c73c3f6a2ea764977d6eaf0a287722d1e2066b4697088c552e342c790f3d2b85"
+SOURCE_DIR="flite-${portVersion}-release"
+PATCHES="flite-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: Mark x86_gcc2 as broken (or maybe fix the parse errors)

Comment (Begasus) [reply app-accessibility/flite/flite-2.1.recipe]: Strike that, the previous patch should have solved that https://github.com/haikuports/haikuports/blob/9157c823aee653229d9da28843252f116e7265fa/app-accessibility/flite/patches/flite-2.0.0.patchset#L53

File: app-accessibility/flite/flite-2.1.recipe [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	flite_tools = $portVersion
+	cmd:flite$secondaryArchSuffix = $portVersion
+	cmd:flite_cmu_time_awb = $portVersion
+	cmd:flite_cmu_us_awb = $portVersion
+	cmd:flite_cmu_us_kal = $portVersion
+	cmd:flite_cmu_us_kal16 = $portVersion
+	cmd:flite_cmu_us_rms = $portVersion
+	cmd:flite_cmu_us_slt = $portVersion
+	cmd:flite_time$secondaryArchSuffix = $portVersion
+	lib:libflite$secondaryArchSuffix = $portVersion compat >= 2.1
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: compat is wrong here and below, try to switch to libVersionCompat (saves changed lines in a future update)

File: app-accessibility/flite/flite-2.1.recipe [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+CHECKSUM_SHA256="c73c3f6a2ea764977d6eaf0a287722d1e2066b4697088c552e342c790f3d2b85"
+SOURCE_DIR="flite-${portVersion}-release"
+PATCHES="flite-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86 x86_gcc2"
+
+PROVIDES="
+	flite$secondaryArchSuffix = $portVersion
+	flite_tools = $portVersion
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: Maybe provide a separate package PROVIDES_tools with the cmd's? (at this moment missing $secondaryArchSuffix, could be fixed by using commandBinDir)

**Comment by Begasus** [comment]
You can add a "defineDebugInfoPackage" to strip the libraries and binaries too

File: app-accessibility/flite/patches/flite-2.1.patchset [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+diff --git a/config/project.mak b/config/project.mak
+index 1d655d9..a1d0729 100644
+--- a/config/project.mak
++++ b/config/project.mak
+@@ -40,4 +40,4 @@ PROJECT_VERSION = 2.1
+ PROJECT_DATE = Dec 2017
+ PROJECT_STATE = release
+ 
+-PROJECT_SHLIB_VERSION = 1
++PROJECT_SHLIB_VERSION = 2
```
Comment (Begasus) [inline app-accessibility/flite/patches/flite-2.1.patchset]: @korli this is done the first time too, looking at their github page I see they still use 1 here, shouldn't we keep it at 1 also then?

File: app-accessibility/flite/flite-2.1.recipe (hunk 4 above)
Comment (alwayslivid) [reply app-accessibility/flite/flite-2.1.recipe]: Added commandBinDir.

File: app-accessibility/flite/flite-2.1.recipe [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+	1999-2014 Language Technologies Institute, Carnegie Mellon University
+	2001 Cepstral, LLC
+	2001 Sun Microsystems, Inc.
+	2003-2004 Nagoya Institute of Technology
+	2003-2004 Nara Institute of Science and Technology
+	2008-2012 University of Illinois at Urbana-Champaign"
+# Although individually many files have different licenses, as a whole this
+# license covers all the files.
+LICENSE="Flite"
+REVISION="3"
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: As this is an updated version revision should be 1

File: app-accessibility/flite/flite-2.1.recipe [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+SOURCE_URI="http://www.festvox.org/flite/packed/flite-2.1/flite-${portVersion}-release.tar.bz2"
+CHECKSUM_SHA256="c73c3f6a2ea764977d6eaf0a287722d1e2066b4697088c552e342c790f3d2b85"
+SOURCE_DIR="flite-${portVersion}-release"
+PATCHES="flite-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86 x86_gcc2"
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: Suggestion:
```diff
-ARCHITECTURES="?x86_gcc2 x86 x86_64"
-SECONDARY_ARCHITECTURES="x86 x86_gcc2"
+ARCHITECTURES="?x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="?x86_gcc2 x86"
```

File: app-accessibility/flite/flite-2.1.recipe [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+	flite$secondaryArchSuffix = $portVersion
+	flite_tools = $portVersion
+	cmd:flite$secondaryArchSuffix = $portVersion
+	cmd:flite_cmu_time_awb = $portVersion
+	cmd:flite_cmu_us_awb = $portVersion
+	cmd:flite_cmu_us_kal = $portVersion
+	cmd:flite_cmu_us_kal16 = $portVersion
+	cmd:flite_cmu_us_rms = $portVersion
+	cmd:flite_cmu_us_slt = $portVersion
+	cmd:flite_time$secondaryArchSuffix = $portVersion
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: No need for $secondaryArchSuffix, missing $commandSuffix

File: app-accessibility/flite/flite-2.1.recipe [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+	devel:libflite_cmu_us_kal$secondaryArchSuffix = $libVersionCompat
+	devel:libflite_cmu_us_kal16$secondaryArchSuffix = $libVersionCompat
+	devel:libflite_cmu_us_rms$secondaryArchSuffix = $libVersionCompat
+	devel:libflite_cmu_us_slt$secondaryArchSuffix = $libVersionCompat
+	devel:libflite_cmulex$secondaryArchSuffix = $libVersionCompat
+	devel:libflite_usenglish$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	flite$secondaryArchSuffix == $portVersion base
+	haiku${secondaryArchSuffix}_devel
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: No need for haiku_devel here
```diff
-	haiku${secondaryArchSuffix}_devel
```

File: app-accessibility/flite/flite-2.1.recipe [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: No need for makefile_engine
```diff
-	makefile_engine
```

File: app-accessibility/flite/flite-2.1.recipe [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+	"
+
+defineDebugInfoPackage flite$secondaryArchSuffix \
+	$binDir/flite_cmu_time_awb \
+	$binDir/flite_cmu_us_awb \
+	$binDir/flite_cmu_us_kal \
+	$binDir/flite_cmu_us_kal16 \
+	$binDir/flite_cmu_us_rms \
+	$binDir/flite_cmu_us_slt \
+	$binDir/flite_time \
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: Use $commandBinDir here

File: app-accessibility/flite/flite-2.1.recipe [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+	$libDir/libflite.so.2 \
+	$libDir/libflite_cmu_indic_lang.so.2 \
+	$libDir/libflite_cmu_indic_lex.so.2 \
+	$libDir/libflite_cmu_time_awb.so.2 \
+	$libDir/libflite_cmu_us_awb.so.2 \
+	$libDir/libflite_cmu_us_kal16.so.2 \
+	$libDir/libflite_cmu_us_slt.so.2 \
+	$libDir/libflite_cmulex.so.2 \
+	$libDir/libflite_usenglish.so.2
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: libVersion is not 2

File: app-accessibility/flite/patches/flite-2.1.patchset [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+-- 
+2.19.0
+
+
+From 95289bfce0ebaac38815569a4910edacefcc0b8a Mon Sep 17 00:00:00 2001
+From: Timothy Gu <timothygu99@gmail.com>
+Date: Wed, 24 Dec 2014 22:39:36 +0000
+Subject: Separate code and declaration
```
Comment (Begasus) [inline app-accessibility/flite/patches/flite-2.1.patchset]: You need 1 more patch to make this work for gcc2

File: app-accessibility/flite/patches/flite-2.1.patchset [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+diff --git a/lang/cmu_indic_lex/cmu_indic_lex.c b/lang/cmu_indic_lex/cmu_indic_lex.c
+index e0ae73f..7444472 100644
+--- a/lang/cmu_indic_lex/cmu_indic_lex.c
++++ b/lang/cmu_indic_lex/cmu_indic_lex.c
+@@ -420,6 +420,9 @@ cst_val *cmu_indic_lex_ord_to_phones(const cst_val *ords,
+                                      int cmu_indic_variant_deletes_word_final_schwa,
+                                      const cst_features *feats)
+ {
++    const cst_val *v, *vx;
```
Comment (Begasus) [inline app-accessibility/flite/patches/flite-2.1.patchset]: Check previous patch and see where it was used

File: app-accessibility/flite/flite-2.1.recipe (hunk 12 above)
Comment (Begasus) [reply app-accessibility/flite/flite-2.1.recipe]: libflite_usenglish.so.2 and the others are no files ...

Comment (alwayslivid) [reply app-accessibility/flite/flite-2.1.recipe]: They seem to appear in the "Contents" section of HaikuDepot, as well as the Makefile on the root directory of flite.

File: app-accessibility/flite/patches/flite-2.1.patchset (hunk 5 above)
Comment (korli) [reply app-accessibility/flite/patches/flite-2.1.patchset]: In the previous patch there was one line explaining this. It was: "This way it is consistent with libflite.so.2.0.0"

**Comment by TimothyGu** [comment]
Glad to see that this is getting pushed across the finish line :)

**Comment by Begasus** [comment]
> Glad to see that this is getting pushed across the finish line after my first attempt for Google Code-In 2014 :)

@TimothyGu any idea why this was changed the previous time? 
+PROJECT_SHLIB_VERSION = 2
Looking at their github page they still use 1 there

File: app-accessibility/flite/patches/flite-2.1.patchset (hunk 5 above)
Comment (Begasus) [reply app-accessibility/flite/patches/flite-2.1.patchset]: OK, see it now, build with default 1 gives libflite_*_.awb.so1 ... should be OK to keep then :)

**Comment by Begasus** [comment]
LGTM

**Comment by Begasus** [comment]
@alwayslivid can you remove the previous recipe and patchset while in the process?

File: app-accessibility/flite/flite-2.1.recipe [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+libVersion="2.1"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	flite$secondaryArchSuffix = $portVersion
+	flite_tools$secondaryArchSuffix = $portVersion
```
Comment (Begasus) [inline app-accessibility/flite/flite-2.1.recipe]: There is nothing providing flite_tools, so this line can be removed:
```diff
-	flite_tools$secondaryArchSuffix = $portVersion
```
We could move the cmd's to a seperate tools package, but for now leave them as is

**Comment by Begasus** [comment]
Don't merge just yet :)
@korli, @alaviss  2 questions, should we move the cmd's to a seperate tools package?
And maybe drop the cmd's for secondary architecture?

**Comment by korli** [comment]
I would have left broken for gcc2, this package is only for Qt5 I think.

**Comment by alwayslivid** [comment]
> @alwayslivid can you remove the previous recipe and patchset while in the process?

Sure.

**Comment by alwayslivid** [comment]
> I would have left broken for gcc2, this package is only for Qt5 I think.

@Begasus, can you confirm?

**Comment by Begasus** [comment]
> > I would have left broken for gcc2, this package is only for Qt5 I think.
> 
> @Begasus, can you confirm?
> 
> If that's the case, all x86_gcc2 patches will be rendered useless and will have to be removed as well.

@korli is correct only Qt5 is using it https://github.com/haikuports/haikuports/search?q=lib%3Alibflite&unscoped_q=lib%3Alibflite
But since the patches are in it doesn't harm to keep them, there are other packages providing gcc2 and gcc8 packages and only used by others for gcc8

**Comment by korli** [comment]
> But since the patches are in it doesn't harm to keep them, there are other packages providing gcc2 and gcc8 packages and only used by others for gcc8

The trend should be in the other direction. Either the gcc2 patch can be upstreamed or not. If not, this is more of a burden for us.

**Comment by Begasus** [comment]
> > But since the patches are in it doesn't harm to keep them, there are other packages providing gcc2 and gcc8 packages and only used by others for gcc8
> 
> The trend should be in the other direction. Either the gcc2 patch can be upstreamed or not. If not, this is more of a burden for us.

Agreed, we could try to upstream as they have the sources on github, or drop the patches and loose support for gcc2 (which wont be that bad either)

**Comment by Begasus** [comment]
Don't think upstream is looking out for gcc2 patches :)

**Pull request closed without merging**